.buildkite/publish-pod.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash -eu
+
+PODSPEC_PATH="Automattic-Tracks-iOS.podspec"
+SLACK_WEBHOOK=$PODS_SLACK_WEBHOOK
+
+echo "--- :rubygems: Setting up Gems"
+install_gems
+
+echo "--- :cocoapods: Publishing Pod to CocoaPods CDN"
+publish_pod $PODSPEC_PATH
+
+echo "--- :slack: Notifying Slack"
+slack_notify_pod_published $PODSPEC_PATH $SLACK_WEBHOOK
```

Comment:
It _could_ be put in the same place, but it's somewhat abstraction-resistant.

Two reasons for this:
- Not all Pods have the same steps (some upload to our private specs repo, others don't)
- There are several variables that are referenced from the secrets storage – rather than relying on magic values being set at the lower levels of the abstraction, for the most part I'd prefer to pass them as arguments to hopefully make debugging simpler. 

I think that once we're using SwiftPM for all of our projects and can drop the private specs repo, we may be able to consolidate this a bit more easily?